### PR TITLE
[IO] Add Parquet Dataset Support.

### PR DESCRIPTION
**Pull request by A-Wanderer**

### CODE REVIEW

File: tensorflow/core/ops/parquet_ops.cc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =============================================================================
+#include "tensorflow/core/framework/common_shape_fns.h"
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/framework/shape_inference.h"
+
+namespace tensorflow {
+
+REGISTER_OP("ParquetTabularDataset")
```
Comment (2sin18) [inline tensorflow/core/ops/parquet_ops.cc:21]: Please rename ops to prevent op name conflicts with HybridBackend

File: tensorflow/workspace.bzl [hunk 2]
```diff
@@ -1077,6 +1077,153 @@ def tf_repositories(path_prefix = "", tf_repo_name = ""):
...
 
+    http_archive(
+        name = "arrow",
+        build_file = "//third_party:arrow.BUILD",
+        patch_cmds = [
+            # TODO: Remove the fowllowing once arrow issue is resolved.
+            """sed -i.bak 's/type_traits/std::max<int16_t>(sizeof(int16_t), type_traits/g' cpp/src/parquet/column_reader.cc""",
+            """sed -i.bak 's/value_byte_size/value_byte_size)/g' cpp/src/parquet/column_reader.cc""",
+        ],
+        sha256 = "a27971e2a71c412ae43d998b7b6d06201c7a3da382c804dcdc4a8126ccbabe67",
```
Comment (2sin18) [inline tensorflow/workspace.bzl:1088]: Attention: Arrow 4.x is incompatible with Apsara libraries. Compile DeepRec with Apsara should disable the capabilities.

**Review by 2sin18 — CHANGES_REQUESTED** [review]

**Review by liutongxuan — APPROVED** [review]

File: tensorflow/workspace.bzl (hunk 2 above)
Comment (liutongxuan) [reply tensorflow/workspace.bzl:1088]: 开源版本暂时不考虑飞天，咱们内部版本会有修改。

Comment (2sin18) [reply tensorflow/workspace.bzl:1088]: @A-Wanderer No, you should add a compiler macro to disable the library if DeepRec must be compiled with apsara libraries.

**Review by 2sin18 — APPROVED** [review]

**Review by liutongxuan — APPROVED** [review]

**Review by liutongxuan — APPROVED** [review]

**Review by liutongxuan — APPROVED** [review]

**Review by liutongxuan — APPROVED** [review]

**Pull request merged**